### DISCUSSION TITLE
Can I use Float32 with hls4ml?

### DISCUSSION THREAD

**Discussion by AnouarITI** [fastmachinelearning/hls4ml#659] [Q&A] [answered]
I have a specific use case where I should use Float32 with a small model. Can I force hls4ml to generate an hls project using floats for my model?

**Comment by vloncar** [comment, marked as answer]
Not at the moment. This used to be possible with some caveats, but was not implemented in the newer codebase. Perhaps in the future. What you can do is manually change the precision data types in the generated `defines.h` and `parameters.h` files, but you may run into issues with some layers (pooling comes to mind, perhaps others). I hope you have a strong incentive for doing this because floating-point operations on FPGAs come with lots of issues related to latency. Quantization is the way to go.
